src/search.cpp

```diff
@@ -421,9 +427,24 @@ void Search::Worker::iterative_deepening() {
         if (!threads.stop)
             completedDepth = rootDepth;
 
-        if (rootMoves[0].pv[0] != lastBestMove)
+        // We make sure not to pick an unproven mated-in score,
+        // in case this thread prematurely stopped search (aborted-search).
+        if (threads.abortedSearch && rootMoves[0].score != -VALUE_INFINITE
+            && rootMoves[0].score <= VALUE_TB_LOSS_IN_MAX_PLY)
         {
-            lastBestMove      = rootMoves[0].pv[0];
+            // Bring the last best move to the front for best thread selection.
+            std::sort(
```

Comment:
okay sorry for the earlier suggestion, I don't think `sort` is right here. It is unstable, but even stable_sort, looks wrongish, the implementation does more than the comment wants to achieve. So I suggest we go for the rotate approach, just put this into `misc.h`, preferably into a namespace called `Utility`, you'll need to add `#include <alogrithm>` too.

```cpp
template <typename T, typename Predicate>
void move_to_front(std::vector<T> &vec, Predicate pred) {
  auto it = std::find_if(vec.begin(), vec.end(), pred);

  if (it != vec.end()) {
    std::rotate(vec.begin(), it, it + 1);
  }
}
```